client/keys/add.go

```diff
@@ -208,11 +208,16 @@ func runAddCmd(ctx client.Context, cmd *cobra.Command, args []string, inBuf *buf
 
 	// If we're using ledger, only thing we need is the path and the bech32 prefix.
 	if useLedger {
-		legacyHdPath, _ := cmd.Flags().GetBool(flagLegacyHdPath)
-		if legacyHdPath {
-			coinType = sdk.CoinType
-		} else {
-			coinType = DefaultLedgerCoinType
+
+		// manual coin type should work if you really want to use it
+		if coinType != sdk.GetConfig().GetCoinType() {
+			legacyHdPath, _ := cmd.Flags().GetBool(flagLegacyHdPath)
+
+			if legacyHdPath {
+				coinType = sdk.CoinType
+			} else {
+				coinType = DefaultLedgerCoinType
+			}
```

Comment:
My twisted mind.

The logic is: If you're using a custom coin type -> use that
Otherwise; use either --legacy-hd-path for 118 or nothing for 529.

I don't think there are many other chains that are supporting multiple coin types with ledger